Source/MLX/MLXArray+Indexing.swift

```diff
@@ -521,4 +353,1209 @@ extension MLXArray {
         }
     }
 
+    /// General array indexing with encoded operations.
+    ///
+    /// Typically used via the general array indexing subscript:
+    ///
+    /// ```swift
+    /// array[0, .ellipsis, 0 ..< 5]
+    /// ```
+    ///
+    /// ### See Also
+    /// - <doc:indexing>
+    subscript(
+        operations operations: [MLXArrayIndexOperation],
+        stream stream: StreamOrDevice = .default
+    ) -> MLXArray {
+        get {
+            switch operations.count {
+            case 0:
+                return self
+            case 1:
+                return getItem(src: self, operation: operations[0], stream: stream)
```

Comment:
This matches the python implementation -- there are some optimized implementations for 1 operation.  There might be some use for some other simple patterns like `[.ellipsis, 0]` etc. that we could consider if it was useful.